### PR TITLE
[MillePede] Add optional geometry parameter to `setup()` for dynamic loading of geometries

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

For Phase-2 style alignment campaigns, one needs to load the geometry from XML and not from DB.
In this PR I add an optional geometry parameter to the `setup()` fuction for dynamic loading of geometries:

- Added the 'geometry' parameter to the setup function
- Loads default geometry if the parameter is not provided
- Dynamically loads geometry modules based on the 'Extended<X>' format
- Includes error handling for invalid formats or loading failures

#### PR validation:

`scram b runtests use-ibeos` runs fine.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, to be backported.

### CODE REVIEW

**Comment by mmusich** [comment]
Cc: @henriettepetersen @TomasKello

**Comment by mmusich** [comment]
hold
   * pending validation from alignment group

**Comment by mmusich** [comment]
unhold
  * cross-checked by Tracker Alignment group

File: Alignment/MillePedeAlignmentAlgorithm/python/alignmentsetup/GeneralSetup.py [hunk 1]
```diff
@@ -1,29 +1,54 @@
...
-    process.load('Configuration.Geometry.GeometryRecoDB_cff')
+
+    # Load the geometry
+    # --------------------------------------------------------------------------
+    if geometry == "":
+        # Default geometry
+        print(f"Using Geometry from DB")
+        process.load('Configuration.Geometry.GeometryRecoDB_cff')
+    else:
+        # Check if the geometry string matches the format "Extended<X>", e.g. Extended2026D110
```
Comment (perrotta) [inline Alignment/MillePedeAlignmentAlgorithm/python/alignmentsetup/GeneralSetup.py:33]: `Extended2026D110` has become by now `ExtendedRun4D110`, but ok: this is just an example in the comment field that explains the usage, not meant ton e correspond exactly to an existing geometry

**Comment by perrotta** [comment]
please test
(tests must be refreshed)

**Comment by perrotta** [comment]
please test
(> 1 month old tests must be refreshed)

**Comment by perrotta** [comment]
+alca

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**